src/interfaces/plugins.ts

```diff
@@ -679,17 +696,30 @@ module.exports = (theApp: any) => {
     })
 
     if (typeof plugin.registerWithRouter == 'function') {
-      plugin.registerWithRouter(router)
       if (typeof plugin.getOpenApi == 'function') {
         app.setPluginOpenApi(plugin.id, plugin.getOpenApi())
       }
     }
-    app.use(backwardsCompat('/plugins/' + plugin.id), router)
 
     if (typeof plugin.signalKApiRoutes === 'function') {
       app.use('/signalk/v1/api', plugin.signalKApiRoutes(express.Router()))
     }
   }
+
+  function doRegisterWithRouter(app: any, plugin: PluginInfo, router: Router) {
+    router.use((req, res, next) => {
+      const stopHandlers = onStopHandlers[plugin.id]
+
+      if (stopHandlers.length > 0) next()
+      else res.status(500).send(plugin.id + ' has stopped')
+    })
+
+    if (typeof plugin.registerWithRouter === 'function') {
+      plugin.registerWithRouter(router)
```

Comment:
Add a debug call here, so that a plugin author can verify their routing gets activated?